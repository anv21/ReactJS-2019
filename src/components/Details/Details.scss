$font-size: 18px;

.details_wrapper {
    display: flex;
    position: relative;
    align-items: flex-start;
    justify-content: flex-start;
    font-size: $font-size;
    margin-bottom: 50px;

    .details_search_button {
        position: absolute;
        right: 0;
        top: -20px;
        border: none;
        border-radius: 5px;
        background-color: #fff;
        color: #f55263;
        text-transform: uppercase;
        font-weight: bold;
        padding: 10px 30px;
        cursor: pointer;

        &:hover {
            color: #fff;
            background-color: #f55263;
        }
    }

    .details_image  {
        height: 400px;
        padding-top: 30px;
        margin-right: 50px;
    }
}

.details_desctiption {
    color: #ffffff;

    .details_titile_container {
        display: flex;
        align-items: center;
        padding: 40px 0 20px 0;

        .details_title {
            font-size: 40px;
            font-weight: bold;
            color: #f55263;
            padding-right: 5px;
        }

        .details_vote_average {
            font-size: 110% $font-size;
            color: #84ba67;
            border: 1px solid #84ba67;
            border-radius: 28px;
            width: 40px;
            height: 40px;
            line-height: 40px;
            text-align: center;
        }
    }
}

.details_date_duration {
    padding-top: 20px;

    .details_year {
        margin-right: 25px;
    }
}

.details_overview {
    margin: 50px 0;
}