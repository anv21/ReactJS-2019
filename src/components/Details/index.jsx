import React from 'react';
import PropTypes from 'prop-types';
import './Details.scss';

class Details extends React.Component {
    goToSearchPage = () => {
        this.props.goToSearchPage();
    }

    render() {
        const {item} = this.props;
        const {poster_path, title, vote_average, tagline, release_date, runtime, overview} = item;

        return (
            <div className="details_wrapper">
                <button 
                    className="details_search_button"
                    onClick={this.goToSearchPage}>search
                </button>
                <img 
                    className="details_image"
                    src={poster_path}
                />
                <div className="details_desctiption">
                    <div className="details_titile_container">
                        <span className="details_title">{title}</span>
                        <span className="details_vote_average">{vote_average}</span>
                    </div>
                    <span className="details_tagline">{tagline}</span>
                    <div className="details_date_duration">
                        <span className="details_year">{release_date.slice(0, release_date.indexOf('-'))}</span>
                        <span className="details_runtime">{runtime} min</span>
                    </div>
                    <div className="details_overview">{overview}</div>
                </div>
            </div>
        )
    }
}

Details.propTypes = {
    poster_path: PropTypes.string,
    title: PropTypes.string,
    vote_average: PropTypes.number,
    tagline: PropTypes.string,
    release_date: PropTypes.string,
    runtime: PropTypes.number,
    overview: PropTypes.string,
    goToSearchPage: PropTypes.func
};

Details.defaultProps = {
    poster_path: '',
    title: '',
    vote_average: 0,
    tagline: '',
    release_date: '',
    runtime:  null,
    overview: '',
    goToSearchPage: null
};

export default Details;