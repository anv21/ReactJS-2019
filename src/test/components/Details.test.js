import React from 'react';
import { shallow } from 'enzyme';

import Details from '../../components/Details/index.jsx';

describe('Details', () => {
    const mockGetItems = jest.fn();
    const item = {
        "id": 447365,
        "title": "Guardians of the Galaxy Vol. 3",
        "tagline": "Life finds a way",
        "vote_average": 0,
        "vote_count": 9,
        "release_date": "2020-05-01",
        "poster_path": "https://image.tmdb.org/t/p/w500/ldoY4fTZkGISMidNw60GHoNdgP8.jpg",
        "overview": "The third film based on Marvel's Guardians of the Galaxy.",
        "budget": 0,
        "revenue": 0,
        "genres": [
          "Action",
          "Adventure",
          "Science Fiction"
        ],
        "runtime": 93
    };
    const details = shallow(<Details getItems={mockGetItems} item={item}/>);

    it('should be like snapshot', () => {
        expect(details).toMatchSnapshot();
    });

    it('should call getItems function on DetailsSearchButton click', () => {
        details.find('.details_search_button').simulate('click');
        expect(mockGetItems.mock.calls.length).toBe(1);
    });
});