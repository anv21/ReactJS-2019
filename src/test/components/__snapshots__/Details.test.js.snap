// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details should be like snapshot 1`] = `ShallowWrapper {}`;
